### PR TITLE
Change package name

### PR DESCRIPTION
**Pull request by Guseyn**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ gradle.properties
 out/
 rewrite-java/src/main/gen/
 .idea/
+rewrite.iml
```
Comment (jkschneider) [inline .gitignore:8]: This is an artifact of the old IntelliJ installation that you had. No newer IntelliJ versions should generate this file.

Comment (Guseyn) [reply .gitignore]: @jkschneider thanks, I removed it

**Comment by sambsnyd** [comment]
The more involved part will be updating all the type references and import statements to have the new package. Plus making sure the files actually get moved to their new home, something which should work but I don't know if we've exercised

**Comment by Guseyn** [comment]
> The more involved part will be updating all the type references and import statements to have the new package. Plus making sure the files actually get moved to their new home, something which should work but I don't know if we've exercised

@sambsnyd That totally makes sense. What about to rename it to “SimplyChangePackageName” for the start since I think it’s needed for some cases where we need just to change package name. It’d be a good start. And then create complete version of it with the current name: «ChangePackageName». Although I am not sure is this a good approach in terms of design.
Anyway I am going to work on it, just probably need more information about how it’s better to implement. So any suggestions are welcome.

**Comment by sambsnyd** [comment]
Hey @Guseyn, sorry I forgot to respond to this for so long.
`andThen()`ing a `ChangeType` should handle most of updating type references and imports to the new package name.

As for physically moving the files... let me check. `RefactorVisitor.generate()` can be used to create new files, but I'm not actually sure if we have a mechanism for deleting/moving files yet.

**Comment by sambsnyd** [comment]
Returning `null` from `visitCompilationUnit` should result in the file being deleted. 
So what I suggest is: When you visit the compilation unit to be moved to a new package, store it off to the side and return null.
In `generate()` return the compilation unit you stored off to the side. Be sure to set the appropriate package and corresponding sourcePath on the modified compilation unit. 

That _should_ do the trick. Let me know how it works out.

**Comment by sambsnyd** [comment]
I'm going to close this since there hasn't been any activity in a while and we've made major, incompatible changes to the core APIs. Would be happy to accept a 7.x compatible version.

**Pull request closed without merging**